src/strongbox/curseforge_api.clj

```diff
@@ -15,102 +16,120 @@
   [path string?, & args (s/* any?)]
   (str curseforge-api (apply format path args)))
 
-(defn latest-versions-by-gameVersionFlavor
-  "given a curseforge-api result, returns a map of release data.
-  uses :gameVersionFlavor which, while always present, is inaccurate as a release may support both classic and retail"
-  [api-result]
-
-  ;; 'latestFiles' :
-  ;; - gameVersionFlavour: that is either "wow_retail" or "wow_classic"
-  ;; - fileStatus: 4 means ... ?
-  ;; - releaseType: 1 is regular/stable, 2 is beta, 3 is alpha
-  ;; - exposeAsAlternative: nil is no, true is yes
-  ;;    - see Recount. there is a '-nolib' alternative available.
-
-  ;; we want to say "give me the latest stable release of the wow classic track of this addon
-  ;; for now we're going to ignore anything that isn't a releaseType of 1 (stable)
-
-  (let [latest-files (:latestFiles api-result)
-
-        ;; results appear sorted, but lets be sure as we'll be taking the first
-        desc (comp - compare) ;; most to least recent (desc)
-        latest-files (sort-by :fileDate desc latest-files)
-
-        ;; stable releases only, for now
+;; addon expansion
+
+(defn-spec release-download-url (s/or :ok ::sp/url, :error nil?)
+  [release map?]
+  (try
+    (let [project-file-id (-> release :projectFileId str)
+          offset (- (count project-file-id) 3)
+          bit1 (-> project-file-id (.substring 0 offset))
+          bit2 (-> project-file-id (.substring offset))]
+      (format "https://edge.forgecdn.net/files/%s/%s/%s" bit1 bit2 (:projectFileName release)))
+    (catch java.lang.StringIndexOutOfBoundsException e
+      (warn (format "failed to construct a download url for release '%s'" (:projectFileName release))))))
+
+(defn-spec rename-identical-releases ::sp/list-of-maps
+  [release-list ::sp/list-of-maps]
+  (let [count-occurances (fn [accumulator-map m]
+                           (let [key :projectFileName]
+                             (update accumulator-map (get m key) (fn [x] (inc (or x 0))))))
+        occurances (reduce count-occurances {} release-list)
+        get* #(get %2 %1)
+        rename-release (fn [release]
+                         (assoc release :-unique-name
+                                (if (-> release :projectFileName (get* occurances) (> 1))
+                                  (let [[name _] (fs/split-ext (:projectFileName release))
+                                        pid (:projectFileId release)]
+                                    (format "%s--%s" name pid))
+                                  (:projectFileName release))))]
+    (mapv rename-release release-list)))
+
+(defn-spec older-releases :addon/release-list
+  "releases under `:gameVersionLatestFiles` appear to be the most recent release by `fileType` (stability)
```

Comment:
"these are releases under `:gameVersionLatestFiles` that are the most recent release by `fileType` (stability) and by game/interface version (8.0.3 etc)."